src/server/scripts/KulTiras/WaycrestManor/boss_lord_and_lady_waycrest.cpp

```diff
@@ -0,0 +1,554 @@
+};
+
+enum LordAndLadyWaycrestActions
+{
+    ACTION_HEAL_LORD_WAYCREST = 1,
+    ACTION_HEAL_AND_JOIN_FIGHT
+};
+
+constexpr Position LordWaycrestCombatPosition = { -549.781f, -260.094f, 185.267f };
+
+// 131527 - Lord Waycrest
+struct boss_lord_waycrest : public BossAI
+{
+    boss_lord_waycrest(Creature* creature) : BossAI(creature, DATA_LORD_WAYCREST), _timesHealed(0), _healed(false) { }
+
+    void Reset() override
+    {
+        events.Reset();
+        _timesHealed = 0;
+        _healed = false;
+    }
+
+    void KilledUnit(Unit* victim) override
+    {
+        if (!victim->IsPlayer())
+            return;
+
+        Talk(SAY_SLAY);
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        instance->SetBossState(DATA_LORD_AND_LADY_WAYCREST, FAIL);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_VIRULENT_PATHOGEN_DAMAGE);
+
+        _EnterEvadeMode();
+        _DespawnAtEvade();
+    }
+
+    void DamageTaken(Unit* /*attacker*/, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
+    {
+        if (me->HealthBelowPctDamaged(30, damage))
+        {
+            if (!_healed && _timesHealed < 2)
+            {
+                if (Creature* ladyWaycrest = instance->GetCreature(DATA_LADY_WAYCREST))
+                {
+                    ladyWaycrest->AI()->DoAction(ACTION_HEAL_LORD_WAYCREST);
+                    _timesHealed++;
+                    _healed = true;
+                }
+
+                events.ScheduleEvent(EVENT_RESET_HEAL, 3s);
+            }
+            else if (!_healed && _timesHealed == 2)
+            {
+                if (Creature* ladyWaycrest = instance->GetCreature(DATA_LADY_WAYCREST))
+                {
+                    ladyWaycrest->AI()->DoAction(ACTION_HEAL_AND_JOIN_FIGHT);
+                    _healed = true;
+                }
+            }
+        }
+    }
+
+    void JustEngagedWith(Unit* /*who*/) override
+    {
+        DoCastSelf(SPELL_LORD_WAYCREST_DUMMY_ABSORB);
+        me->GetMotionMaster()->MoveJumpWithGravity(LordWaycrestCombatPosition, 20.0f, 15.1852f);
+
+        instance->SetBossState(DATA_LORD_AND_LADY_WAYCREST, IN_PROGRESS);
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me);
+
+        if (Creature* ladyWaycrest = instance->GetCreature(DATA_LADY_WAYCREST))
+            ladyWaycrest->AI()->DoZoneInCombat();
+
+        events.ScheduleEvent(EVENT_WASTING_STRIKE, 7200ms);
+        events.ScheduleEvent(EVENT_VIRULENT_PATHOGEN, 10800ms);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        if (Creature* ladyWaycrest = instance->GetCreature(DATA_LADY_WAYCREST))
+        {
+            if (ladyWaycrest->IsAlive())
+                return;
+            else
+                instance->SetBossState(DATA_LORD_AND_LADY_WAYCREST, DONE);
+        }
+
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
```

Comment:
you might still want to do this, even when lady is still alive (removes boss frame)